### ISSUE TITLE
Found duplicate package in environment

### ISSUE THREAD

**Issue by amartani** [astral-sh/uv#1848]
In one of my pre-existing venvs, any `uv pip` command that I run results in the following error:

```
$ uv pip install mypy
error: Failed to list installed packages
  Caused by: Found duplicate package in environment: setuptools (/home/aurelia/.pyenv/versions/3.9.11/envs/aurelia/lib/python3.9/site-packages/setuptools-58.1.0.dist-info vs. /home/aurelia/.pyenv/versions/3.9.11/envs/aurelia/lib/python3.9/site-packages/setuptools-67.6.0.dist-info)
```

This is just one example command, but any other `uv pip install` / `uv pip uninstall` / `uv pip sync` on this venv results in the same error. `pip` seems to work fine in this state and believes only `67.6.0` is installed:

```
$ pip list | grep setuptools
setuptools                                        67.6.0
types-setuptools                                  67.6.0.0
```

And similarly other `pip install` / `pip uninstall` / `pip-sync` commands still work on this venv.

I don't know how this venv ended up in this state. I found https://github.com/astral-sh/uv/issues/1062 while searching for this error, but I have never used `conda` (nor any other non-pip package manager) in this venv.

Using uv version 0.1.6 on Linux arm64 (docker on MacOS).

**Comment by charliermarsh** [comment]
Thanks. I assume removing the older version clears this?

**Comment by amartani** [comment]
Yes. There was actually 2 packages in this state (`pip` and `setuptools`); after removing the 2 wrong `.dist-info` files, the error did not happen again.

**Comment by charliermarsh** [comment]
We could consider making this a warning (rather than an error), and respecting the "highest" version of each.

**Comment by nschloe** [comment]
Just ran into this in gh-actions default env:

```
error: Failed to list installed packages
  Caused by: Found duplicate package in environment: pip
(/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pip-24.0.dist-info
vs.
/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pip-23.0.1.dist-info)
```

**Comment by konstin** [comment]
This is a bug in github actions: https://github.com/actions/setup-python/issues/785

Having a venv in this state is bad because it is impossible to determine which is the correct version that `site-packages/pip` belongs too. E.g. if a user requested `pip==24.0`, is this version installed or is `23.0.1` installed. It is also not specified what other tools will believe about the installed version. Normally i'd argue that the user needs to recreate the venv (we can't determine which files we'd need to remove or add to fix it), but the github actions bug is a show stopper.

**Comment by charliermarsh** [comment]
I’ll take this one.

**Comment by charliermarsh** [comment]
Okay, I looked at pip's behavior. They just do a `listdir` and omit packages they've already seen.

**Comment by charliermarsh** [comment]
This wasn’t included in v0.10.0, the fix has been merged but not released yet.
